### ISSUE TITLE
Feature Request: A warning during a boolean comparison if the variable is a Union type with `None`

### ISSUE THREAD

**Issue by efine-vivodyne** [microsoft/pyright#7852]
**Is your feature request related to a problem? Please describe.**
```python
value: int | None = None
if not value:
    do_something
```

for situations where `None` is possible and forms of the other variable type could also evaluate to booleans, bugs can slip in.  If someone meant the check to just be 'if the value doesn't equal zero', then `None` would still trigger it.  alternatively, if someone wrote `if value` when trying to ensure that the value was not `None` then it would also evaluate as false if `value==0`.  These are rarely (if ever) anyone's intent with the code.

**Describe the solution you’d like**
A warning if a boolean comparison is being made if the value is a Union type that can also be `None` (`reportAmbiguousBooleanForOptionalValue` ? ...probably too long) (I feel like mypy had something like this, but I couldn't find it by googling).  
This would force someone to either explicitly ignore the rule with an inline directive if they truly meant to do this.  Or it would likely correct a potential bug in their code by having them explicitly check for `None` --- e.g. `if value is not None` instead of just `if value` or `assert value is not None` prior to the `if` statement

**Comment by erictraut** [comment]
This is the first time this feature has been requested, so it's not clear to me that this situation is sufficiently common to merit a new check. I'm also not aware of any such check in mypy or other type checkers, which leads me to believe it's not a common source of errors. 

I'm going to reject the feature request for now, but I'm willing to revisit it in the future if there's a signal from other pyright users that this would be valuable. If you're interested in this feature, please add a "thumbs up" reaction to this issue.
